docs/class-definitions.md

```diff
@@ -0,0 +1,73 @@
+## Class Definitions
```

Comment:
Should we not be generating these from the doc comments on the classes themselves?

Reply:
I proposed that [here](https://ably-real-time.slack.com/archives/C04B5EKDUKC/p1677769644970309?thread_ts=1677769183.776059&cid=C04B5EKDUKC)

Reply:
replied in slack